### PR TITLE
boards: arm: rpi_pico: add jlink runner conf

### LINKED ISSUE #48194: Support J-Link debugger for RaspberryPi Pico
This issue derives from #45137.

@MeisterBob provided a short code to support the J-Link debugger.

https://github.com/zephyrproject-rtos/zephyr/issues/45137#issuecomment-1182826277

I think it is good to add after #45137 is merged.

### PR DESCRIPTION
**Pull request by MeisterBob**
Add jlink configuration to rpi_pico

closes #48194

### CODE REVIEW

File: boards/arm/rpi_pico/board.cmake [hunk 1]
```diff
@@ -26,3 +26,6 @@ board_runner_args(openocd --cmd-pre-init "source [find target/rp2040.cfg]")
 board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
 
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+
+board_runner_args(jlink "--device=RP2040_M0_0")
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
```
Comment (soburi) [inline boards/arm/rpi_pico/board.cmake:31]: It is preferable to put include statements at the bottom of the file.

Such like ...
```
@@ -25,4 +25,7 @@ board_runner_args(openocd --cmd-pre-init "source [find target/rp2040.cfg]")
 # But if not so, set 2000 to adapter speed.
 board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
+board_runner_args(jlink "--device=RP2040_M0_0")
+
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)

```

Comment (soburi) [inline boards/arm/rpi_pico/board.cmake:31]: It is preferable to put include statements at the bottom of the file.

Such like ...
```
@@ -25,4 +25,7 @@ board_runner_args(openocd --cmd-pre-init "source [find target/rp2040.cfg]")
 # But if not so, set 2000 to adapter speed.
 board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
+board_runner_args(jlink "--device=RP2040_M0_0")
+
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)

```

**Review by soburi — CHANGES_REQUESTED** [review]

**Comment by soburi** [comment]
And need to update document.

- Add Flashing section to describe flash with J-Link

- Following line may need to rewrite.
  https://github.com/zephyrproject-rtos/zephyr/blame/main/boards/arm/rpi_pico/doc/index.rst#L176-L191

**Comment by MeisterBob** [comment]
moved the include statement to the bottom of `board.cmake` and added a few lines to the documentation.

File: boards/arm/rpi_pico/doc/index.rst [hunk 2]
```diff
@@ -86,6 +86,12 @@ Flashing
 Using an SWD adapter
 --------------------
 
+You can Flash the rpi_pico with a SEGGER JLink debug probe with
```
Comment (soburi) [inline boards/arm/rpi_pico/doc/index.rst:89]: This is the 'Using an SWD adapter' section, and the JLink debug probe is not an SWD adapter.

End of this section, line
https://github.com/zephyrproject-rtos/zephyr/blame/52fe06495dc2d9bfa394353778c3f211441a65fb/boards/arm/rpi_pico/doc/index.rst#L134-L139
 is mentioned about how to use JLink.
I think it should be replaced with a new section (I think it is good to be titled 'Using J-Link adapter')
and put this command line into it.

File: boards/arm/rpi_pico/doc/index.rst [hunk 3]
```diff
@@ -151,7 +157,9 @@ UF2 file should be drag-and-dropped to the device, which will flash the Pico.
 Debugging
 =========
 
-The SWD interface can also be used to debug the board. To achieve this,
+The SWD interface can also be used to debug the board. To achieve this, you can
+use a SEGGER JLink debug probe and follow the instruction in
```
Comment (soburi) [inline boards/arm/rpi_pico/doc/index.rst:161]: Debugging section also mentions about J-Link using.
https://github.com/zephyrproject-rtos/zephyr/blame/52fe06495dc2d9bfa394353778c3f211441a65fb/boards/arm/rpi_pico/doc/index.rst#L172-L190

We can replace this lines with
```
west debug --runner jlink
```

**Review by soburi — CHANGES_REQUESTED** [review]
I'm sorry to trouble you, but could you please remove the outdated article about J-Link in the documentation and update it?

File: boards/arm/rpi_pico/doc/index.rst (hunk 2 above)
Comment (MeisterBob) [reply boards/arm/rpi_pico/doc/index.rst:89]: I would call the JLink a SWD (and JTAG, SPI, ...) adapter. I could split it up into `Using jlink` and `Using openocd`

Comment (soburi) [reply boards/arm/rpi_pico/doc/index.rst:89]: `I would call the JLink a SWD (and JTAG, SPI, ...) adapter.`
Sorry, This is correct.

` I could split it up into Using jlink and Using openocd`
👍 
It is more kindly for users. Thank you for suggestion.

**Review by yonsch — APPROVED** [review]

**Review by soburi — APPROVED** [review]
LGTM

**Pull request merged**